### PR TITLE
TSContSchedule API renaming

### PR DESCRIPTION
**Pull request by duke8253**
As mentioned in #6577, we're now removing `TSContSchedule`, then renaming `TSContScheduleOnPool` to `TSContSchedule`. Also updated the docs and tests to reflect the change.

### CODE REVIEW

File: doc/developer-guide/api/functions/TSContSchedule.en.rst [hunk 1]
```diff
@@ -28,31 +28,115 @@ Synopsis
...
+The continuation is scheduled for a particular thread selected from a group of similar threads,
+as indicated by :arg:`tp`. If :arg:`contp` already has an thread affinity set, and the thread
+type of thread affinity is the same as :arg:`tp`, then :arg:`contp` will be scheduled on the
+thread specified by thread affinity.
+
+=========================== =======================================================================================
+Pool                        Properties
+=========================== =======================================================================================
+``TS_THREAD_POOL_NET``      Transaction processing threads. Continuations on these threads must not block.
+``TS_THREAD_POOL_TASK``     Background threads. Continuations can perform blocking operations.
```
Comment (ywkaras) [inline doc/developer-guide/api/functions/TSContSchedule.en.rst:51]: NO_BLOCK and BLOCK as aliases or instead of NET and TASK?

File: doc/developer-guide/api/functions/TSContScheduleEvery.en.rst [hunk 2]
```diff
@@ -33,19 +33,20 @@ Synopsis
 Description
 ===========
 
-Schedules :arg:`contp` to periodically run every :arg:`delay` milliseconds in the future.
-This is approximate. The delay will be at least :arg:`delay` but possibly more.
-Resolutions finer than roughly 5 milliseconds will not be effective. :arg:`contp` is
-required to have a mutex, which is provided to :func:`TSContCreate`.
+Schedules :arg:`contp` to run :arg:`every` milliseconds, on a random thread that belongs to
+:arg:`tp`. The :arg:`every` is an approximation, meaning it will be at least :arg:`every`
+milliseconds but possibly more. Resolutions finer than roughly 5 milliseconds will not be
```
Comment (ywkaras) [inline doc/developer-guide/api/functions/TSContScheduleEvery.en.rst:38]: Hmm is the resolution really that consistent across different flavors of Linux, and FreeBSD and macOS?

Comment (duke8253) [reply doc/developer-guide/api/functions/TSContScheduleEvery.en.rst:38]: I'd assume so since this is in the docs way back before I ever touched anything.

File: doc/developer-guide/api/functions/TSContSchedule.en.rst (hunk 1 above)
Comment (duke8253) [reply doc/developer-guide/api/functions/TSContSchedule.en.rst:51]: That's not a bad idea, @SolidWallOfCode  what do you think?

**Comment by bryancall** [comment]
@duke8253 After the change @sudheerv made we are now scheduling the event on the same thread.  This change makes it so that we now would schedule on the net thread pool vs the same thread?  If this is the case, it would be better to schedule on the same thread instead of switching threads.

**Comment by duke8253** [comment]
> @duke8253 After the change @sudheerv made we are now scheduling the event on the same thread. This change makes it so that we now would schedule on the net thread pool vs the same thread? If this is the case, it would be better to schedule on the same thread instead of switching threads.

Not exactly. It will schedule on a thread that matches the type, so the person scheduling must provide a type. If the type is the same as the current thread it's on, then it'll schedule on the current thread.

File: doc/developer-guide/api/functions/TSContSchedule.en.rst (hunk 1 above)
Comment (SolidWallOfCode) [reply doc/developer-guide/api/functions/TSContSchedule.en.rst:51]: A bit too radical of a change, given all the documentation and code that uses those terms.

**Comment by SolidWallOfCode** [comment]
There is an API function to schedule on a specific thread (which I think @keesspoelstra asked for at one summit, and lead to all of this...). Therefore, if what you want it so schedule on the current thread, you can easily do that.

**Comment by SolidWallOfCode** [comment]
I'm impressed by the amount of documentation. Nice job!

File: doc/developer-guide/api/functions/TSContSchedule.en.rst (hunk 1 above)
Comment (ywkaras) [reply doc/developer-guide/api/functions/TSContSchedule.en.rst:51]: Adding aliases would not require changing any of that.

**Comment by duke8253** [comment]
I'm fine with either, I'll revert back to not changing the names then?

**Comment by sudheerv** [comment]
> I don't think we should rename TSContScheduleOnPool() to be TSContSchedule(). The function requires a pool and it makes more sense to have the name TSContScheduleOnPool(). I have mix feeling on getting rid of TSContSchedule()

Yeah, I tend to agree with Bryan -  the naming of TSContScheduleOnPool() and TSContschedule() are more readable  and intuitive to relate to the intended behavior. Even though TSContScheduleOnPool() can indirectly achieve the effect of scheduling on the current thread, it feels a bit too contrived and perhaps a bit needlessly complicated that it’d only do so when the affinity is not set. As a API user, one would much rather have a simple and direct API that says TSContSchedule() that defaults to the current thread. This is actually why I didn’t like reusing TSContScheduleOnPool() naming if we wanted to kill one of the API.

 I dont particularly think having two separate (clearly named) API in this case is confusing or overload. Rather, repurposing one API to support too many combinations seems the more confusing option.

Long story short, while I’m not -1 to condense into a single API and would be okay if everyone else feels that’s the right way to go, I’m definitely +1 on leaving the (2 separate) API as they are in ATS9 (current master).

**Comment by SolidWallOfCode** [comment]
Given what I suggested [a month ago](https://github.com/apache/trafficserver/pull/6577#issuecomment-604720341) I am glad to see @sudheerv and @bryancall have come around to my point of view. We should have `TSContScheduleOnPool` for pool based scheduling and `TSContSchedulOnThread` for thread based scheduling. Scheduling on the current thread is then
```
TSContScheduleOnThread(cont, TSEventThreadSelf());
```
Which I find much clearer than just `TSContSchedule`, which I think should be reserved for purely "automatic" scheduling, in situations where the continuation came from somewhere else and it should be sent back there (via thread affinity).

**Comment by duke8253** [comment]
Everything should be fixed by now.

**Comment by zwoop** [comment]
Reverted from the 9.1.x branch, and restored to Milestone 10.0.0 for now.
